oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedTreeStoreTask.java

```diff
@@ -230,4 +237,68 @@ private void sortAndSaveBatch(NodeStateEntryBatch nseb) throws Exception {
         }
     }
 
+    /**
+     * If there are any, remove properties of bundled nodes (jcr:content/...) from the JSON-encoded node.
+     *
+     * @param path the path
+     * @param value the JSON-encoded node
+     * @return the cleaned JSON
+     */
+    public static String removePropertiesOfBundledNodes(String path, String value) {
+        if (value.indexOf("\"jcr:content/") < 0) {
+            return value;
+        }
+        // possibly the node contains a bundled property, but we are not sure
+        // try to de-serialize
+        NodeStateEntryReader nodeReader = new NodeStateEntryReader(new MemoryBlobStore());
+        try {
+            // the following line will throw an exception if de-serialization fails
+            nodeReader.read(path + "|" + value);
+            // ok it did not: it was a false positive
+            return value;
+        } catch (Exception e) {
+            LOG.warn("Path {} value {}", path, value);
```

Comment:
Could we make this a bit more descriptive ? Unable to de-serialize due to presence of bundled properties.